consumption.tex

```diff
@@ -0,0 +1,50 @@
+
+\subsection{Block Connection Proof}
+
+We will look at two ways of proving block connection to the honestly adopted source chain.
+
+\subsubsection{Federation}
+A federation of $m$ nodes is picked. A federation node is responsible for running the source blockchain software and knowing at every moment the honestly adopted source blockchain. More importantly, it is responsible for transmitting this information to the target blockchain. In order to do this, we define a smart contract on the target blockchain where federation nodes are allowed to submit such information. We call this contract the \textsf{TagRepository}. A federation node encodes the honestly adopted chain into a tag and submits this tag to the \textsf{TagRepository}. Once a majority of votes is achieved, the tag is considered valid for use.
+
+Similar to FlyClient~\cite{flyclient}, we propose encoding the honestly adopted chain by using a Merkle Mountain Range (MMR). Specifically, to generate a tag we take all the chain's block ids in sequence and use them as leaves in a Merkle Mountain Range. We then use the root of that MMR as our encoding.
+
+Knowing this, it is now possible to define \textsf{blockConnection}. This component will be an MMR proof for the fact that that $id(\mathsf{blockHeader}) \in \text{MMR}$ in the form of $\mathsf{blockConnection} = [(x_0, s_0), (x_1, s_1), \dots]$, where $s_i \in \sf\{Left, Right\}$.
+
+An interesting observation is that, because the federation majority is trusted on the target blockchain, we could avoid doing the block header hashing and verification in order to verify block connection. Instead of storing the block id of each block in the MMR, we could store the transaction merkle root of each block, and again use the root of the MMR as the tag. The verifier on the smart contract side would then verify that the \textsf{txIDMerkleRoot} provided exists in the MMR using the \textsf{blockConnection}.
+
+\import{./}{algorithms/alg.verify-event-federation.tex}
+
+\subsubsection{Decentralized}
+We introduce a second way for proving and verifying block connection which does not require a trusted federation. This is possible by making use of NIPoPoWs. The way the proving process works differs quite a bit from the federation, because in the federation there is a trusted source of truth. If a tag is voted by the federation majority, then it is forever true. But if this the verifier does not have this information, then it has to rely on the prover to get it. We will need to assume a model where there exist two provers, one honest and one malicious. The block connection proof can be an NIPoPoW infix proof for the block which includes the burn transaction. Specifically, $\mathsf{blockConnection} = \pi\chi$.
```

Comment:
*the* block connection

Reply:
Cite NIPoPoWs.

Reply:
Avoid informal "quite a bit".

Reply:
"if this the"?

Reply:
"We will need to assume" is strong. Do we have an impossibility result which mandates that this is necessary? No. So we don't know if we will *need* to assume anything. But we will assume. So we can just state "We assume".

Reply:
an NIPoPoW -> a NIPoPoW

Reply:
I feel this paragraph delves too much into NIPoPoWs and assumes expertise there without introducing any of the concepts such as "infix proofs". It will be extremely challenging to read, even for a blockchain expert. I would tone it down a bit.

Reply:
We have not introduced $\pi$ and $\chi$. At least a basic explanation is warranted.